test/models/maintenance_tasks/task_data_test.rb

```diff
@@ -50,6 +51,14 @@ class TaskDataTest < ActiveSupport::TestCase
         task_data.code.each_line.grep(/UpdatePostsTask/).first.squish
     end
 
+    test "#code returns the code source of a Task with a prepended module" do
+      task_klass = "UpdatePostsModulePrependedTask"
+      task_data = TaskData.new("Maintenance::#{task_klass}")
+
+      assert_equal "class #{task_klass} < MaintenanceTasks::Task",
+        task_data.code.each_line.grep(/#{task_klass}/).first.squish
```

Comment:
Also this test errors when the condition is not respected ("NoMethodError: undefined method `squish' for nil:NilClass"), ideally it would fail:

```suggestion
      assert_includes task_data.code,
        "class #{task_klass} < MaintenanceTasks::Task"
```

Reply:
✅ Agreed and updated! I referenced the [existing test](https://github.com/Shopify/maintenance_tasks/blob/main/test/models/maintenance_tasks/task_data_test.rb#L46-L51) of a similar format so I've updated both here.